st3/lsp_utils/npm_client_handler.py

```diff
@@ -172,7 +207,19 @@ def on_initialized(self, client) -> None:
         """
         This method should not be overridden. Use the `on_ready` abstraction.
         """
-        self.on_ready(ApiWrapper(client))
+        api = ApiWrapper(client)
+        self._register_custom_server_event_handlers(api)
+        self.on_ready(api)
 
     def on_ready(self, api: ApiWrapper) -> None:
         pass
+
+    def _register_custom_server_event_handlers(self, api: ApiWrapper) -> None:
+        for _, func in inspect.getmembers(self, predicate=inspect.isroutine):
+            event_name = getattr(func, ApiWrapper.HANDLER_MARK_NOTIFICATION, None)  # type: Optional[str]
+            if event_name:
+                api.on_notification(event_name, func)
+
+            event_name = getattr(func, ApiWrapper.HANDLER_MARK_REQUEST, None)  # type: Optional[str]
+            if event_name:
+                api.on_request(event_name, func)
```

Comment:
I suppose you wouldn't want to decorate the same function with both decorators as that wouldn't work. So should break early.

Reply:
Yes, I think it doesn't make sense to have a handler handles both notification and request. Will change.

Reply:
Resolved as of https://github.com/sublimelsp/lsp_utils/pull/38/commits/179920b79cf7cdac68ffd5ab5cc5d68203f3e7da